grpc/client.go

```diff
@@ -39,12 +39,33 @@ type registry interface {
 	MustRegister(...prometheus.Collector)
 }
 
+// clientMetrics is a struct type used to return registered metrics from
+// `NewClientMetrics`
+type clientMetrics struct {
+	grpcMetrics *grpc_prometheus.ClientMetrics
+	// inFlightRPCs is a labelled gauge that slices by service/method the number
+	// of outstanding/in-flight RPCs.
+	inFlightRPCs *prometheus.GaugeVec
+}
+
 // NewClientMetrics constructs a *grpc_prometheus.ClientMetrics, registered with
 // the given registry, with timing histogram enabled. It must be called a
 // maximum of once per registry, or there will be conflicting names.
-func NewClientMetrics(stats registry) *grpc_prometheus.ClientMetrics {
-	metrics := grpc_prometheus.NewClientMetrics()
-	metrics.EnableClientHandlingTimeHistogram()
-	stats.MustRegister(metrics)
-	return metrics
+func NewClientMetrics(stats registry) clientMetrics {
+	// Create the grpc prometheus client metrics instance and register it
+	grpcMetrics := grpc_prometheus.NewClientMetrics()
+	grpcMetrics.EnableClientHandlingTimeHistogram()
+	stats.MustRegister(grpcMetrics)
+
+	// Create a gauge to track in-flight RPCs and register it.
+	inFlightGauge := prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "in_flight_rpcs",
```

Comment:
I should have thought of this on the last change: Let's prefix this with "grpc" like the other metrics: "grpc_in_flight"

Reply:
:+1: Fixed in 2fb064c

Should I make a separate PR to fix the server interceptor latency histogram name?

Reply:
Yes please.

Reply:
Done! https://github.com/letsencrypt/boulder/pull/3673